scripts/deployment/pip/requirements/production.in

```diff
@@ -2,39 +2,39 @@
 Django==1.11.28
 
 # Django filter
-django-filter==1.1
+django-filter
```

Comment:
Not having version numbers is a no-no in my book and I am sure it is also in yours, so do you mind explaining how the version numbers of dependencies will be calculated to make the builds repeatable?  

A pointer to the relevant Python documentation is good enough.

Reply:
The versions of packages are not changed with each build. We explicitly "request" an update intentionally, when we wish to update packages. But, the calculation of the version numbers is done smartly so that all the packages still remain compatible with each other, without needing us to manually figure them out. 
The README of pip-tools (of which pip-compile is a part of) has some explanation: https://github.com/jazzband/pip-tools/

Reply:
Thanks. So basically the .in files are "compiled" into the requirements.txt files (which are part of the git repo) whenever we manually run the pip-tools thing, right?

Reply:
Yes, there's a helper script `script/devhelpers/update-locked-requirements` that does this compilation. And, the compiled `requirements.txt` files are actually used for the installation. This change came in a previous commit that was reviewed by @zuhdil. I should've asked you for a review on that too. :)